101-datastore-create-blob/prereqs/prereq.azuredeploy.json

```diff
@@ -124,7 +124,7 @@
       "value": "[parameters('storageAccountName')]"
     },
 	"storageAccountAccountKey": {
-      "type": "string",
+      "type": "securestring",
```

Comment:
you won't be able to do this... what you need to do is output the resourceId of the storage account and then use that in the listKeys() function in azuredeploy.json (to avoid passing secrets)

Reply:
There is a use case where customer can create datastore without storageKey or SAS token. If we do this in azuredeploy.json it will always set AccountKey even if customer don't want it. How can we handle this usecase? There is no way to know if customer want to set it or not.

Reply:
Not sure I follow - right now we're passing a secret via outputs, that's a security issue so we're phasing it out of the repo.

If the property is optional, then you can use an empty string to determine the behavior and set the property value to json('null') (if that's what's needed).

If you want to set the property, then just pass the resourceId (or account name and resourcegroup name) into the template and call listKey() there.

That help?

Reply:
Thank you @bmoore-msft , I updated to change the desired behavior.